server/src/modules/plugins/service.ts

```diff
@@ -21,6 +21,12 @@ export class PluginsService implements IPluginsService {
 
   async install(body: CreatePluginDto) {
     const { id, repo } = body;
+    
+    const existingPlugin = await dbTransactionWrap((manager: EntityManager) => {
+      return manager.findOne(Plugin, { where: { pluginId: id } });
```

Comment:
Can we also add a uniqueness constraint at the DB level to avoid duplicates on requests with race conditions?